rasa/shared/utils/common.py

```diff
@@ -66,6 +68,64 @@ def _lazyprop(self):
     return _lazyprop
 
 
+def cached_method(f: Callable[..., Any]) -> Callable[..., Any]:
+    """Caches method calls based on the call's `args` and `kwargs`.
+
+    Works for `async` and `sync` methods. Don't apply this to functions.
+
+    Returns:
+        The return value which the method gives for the first call with the given
+        arguments.
+    """
+
```

Comment:
should we check if the first argument of `f` is `self` to ensure it is a method?

Reply:
Done. I couldn't use `inspect.ismethod` as it all coroutines are functions 💩